### ISSUE TITLE
conans.errors.ConanException: libwebp/1.0.3@bincrafters/stable: 'settings.compiler' value not defined

### ISSUE THREAD

**Issue by apophatique** [anibali/pywebp#20]
Hello, please, help me, i just dont know what to do
I'm trying to install webp in pycharm on windows, but i get this error:

![image](https://user-images.githubusercontent.com/44003762/99079612-63b03780-25ea-11eb-904c-7fcc7db9d940.png)

In this thread (https://stackoverflow.com/questions/64799277/installing-of-webp-conan-errror/64802122#64802122) i've got an advice to install Visual Studio, but it doesnt help. 

I'm trying to install webp for a 3-4 days, please help me, what should i do? 

Thanks

**Comment by anibali** [comment]
Installing Visual Studio (with the C++ build tools and Windows 10 SDK) is correct. However, you must also configure your Conan profile so that it knows to use the compiler you installed.

```
conan profile update "settings.compiler=Visual Studio" default
conan profile update "settings.compiler.version=14" default
pip install webp
```

**Comment by apophatique** [comment]
> Installing Visual Studio (with the C++ build tools and Windows 10 SDK) is correct. However, you must also configure your Conan profile so that it knows to use the compiler you installed.
> 
> ```
> conan profile update "settings.compiler=Visual Studio" default
> conan profile update "settings.compiler.version=14" default
> pip install webp
> ```

Oh, thank you man! It actually works now

**Comment by FeraruSilviuMarian** [comment]
`C:\Users\silve>pip install webp
Collecting webp
  Using cached webp-0.1.2.tar.gz (11 kB)
Requirement already satisfied: cffi>=1.0.0 in c:\users\silve\appdata\local\programs\python\python38-32\lib\site-packages (from webp) (1.14.6)
Requirement already satisfied: Pillow>=4.0.0 in c:\users\silve\appdata\local\programs\python\python38-32\lib\site-packages (from webp) (8.3.2)
Requirement already satisfied: numpy>=1.0.0 in c:\users\silve\appdata\local\programs\python\python38-32\lib\site-packages (from webp) (1.21.2)
Requirement already satisfied: pycparser in c:\users\silve\appdata\local\programs\python\python38-32\lib\site-packages (from cffi>=1.0.0->webp) (2.20)
Building wheels for collected packages: webp
  Building wheel for webp (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'c:\users\silve\appdata\local\programs\python\python38-32\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\silve\\AppData\\Local\\Temp\\pip-install-42g1k2k6\\webp_fb85ab40d19c4d06afd812d2782552ff\\setup.py'"'"'; __file__='"'"'C:\\Users\\silve\\AppData\\Local\\Temp\\pip-install-42g1k2k6\\webp_fb85ab40d19c4d06afd812d2782552ff\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\silve\AppData\Local\Temp\pip-wheel-24nhc3sd'
       cwd: C:\Users\silve\AppData\Local\Temp\pip-install-42g1k2k6\webp_fb85ab40d19c4d06afd812d2782552ff\
  Complete output (85 lines):
  Configuration:
  [settings]
  arch=x86_64
  arch_build=x86_64
  build_type=Release
  compiler=Visual Studio
  compiler.runtime=MD
  compiler.version=14
  os=Windows
  os_build=Windows
  [options]
  [build_requires]
  [env]

  conanfile.txt: Installing package
  Requirements
      libwebp/1.0.3 from 'conancenter' - Cache
  Packages
      libwebp/1.0.3:8353b55c00a74fef4c1485ae46daf301a093802c - Cache

  Installing (downloading, building) binaries...
  libwebp/1.0.3: Already installed!
  conanfile.txt: Generator json created conanbuildinfo.json
  conanfile.txt: Generator txt created conanbuildinfo.txt
  
  ...

build\lib.win32-3.8\_webp.pyd : fatal error LNK1120: 28 unresolved externals
  error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x86\\link.exe' failed with exit status 1120
  Error in atexit._run_exitfuncs:
  Traceback (most recent call last):
    File "C:\Users\silve\AppData\Roaming\Python\Python38\site-packages\colorama\ansitowin32.py", line 59, in closed
      return stream.closed
  ValueError: underlying buffer has been detached
  ----------------------------------------
  ERROR: Failed building wheel for webp
  Running setup.py clean for webp
Failed to build webp
Installing collected packages: webp
    Running setup.py install for webp ... error
    ERROR: Command errored out with exit status 1:


...


Complete output (85 lines):
    Configuration:
    [settings]
    arch=x86_64
    arch_build=x86_64
    build_type=Release
    compiler=Visual Studio
    compiler.runtime=MD
    compiler.version=14
    os=Windows
    os_build=Windows
    [options]
    [build_requires]
    [env]

    conanfile.txt: Installing package
    Requirements
        libwebp/1.0.3 from 'conancenter' - Cache
    Packages
        libwebp/1.0.3:8353b55c00a74fef4c1485ae46daf301a093802c - Cache

    Installing (downloading, building) binaries...
    libwebp/1.0.3: Already installed!
    conanfile.txt: Generator txt created conanbuildinfo.txt
    conanfile.txt: Generator json created conanbuildinfo.json
    conanfile.txt: Generated conaninfo.txt
    conanfile.txt: Generated graphinfo
    running install
    running build
    running build_py
    creating build
    creating build\lib.win32-3.8
    creating build\lib.win32-3.8\webp
    copying webp\__init__.py -> build\lib.win32-3.8\webp
    creating build\lib.win32-3.8\webp_build
    copying webp_build\builder.py -> build\lib.win32-3.8\webp_build
    copying webp_build\__init__.py -> build\lib.win32-3.8\webp_build
    copying webp_build\cdef.h -> build\lib.win32-3.8\webp_build
    copying webp_build\source.c -> build\lib.win32-3.8\webp_build
    running build_ext
    generating cffi module 'build\\temp.win32-3.8\\Release\\_webp.c'
    creating build\temp.win32-3.8
    creating build\temp.win32-3.8\Release
    building '_webp' extension
    creating build\temp.win32-3.8\Release\build
    creating build\temp.win32-3.8\Release\build\temp.win32-3.8
    creating build\temp.win32-3.8\Release\build\temp.win32-3.8\Release
    C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133


...


    _webp.obj : error LNK2001: unresolved external symbol _WebPPictureImportRGBA
    _webp.obj : error LNK2001: unresolved external symbol _WebPEncode
    _webp.obj : error LNK2001: unresolved external symbol _WebPInitDecoderConfigInternal
    _webp.obj : error LNK2001: unresolved external symbol _WebPAnimEncoderAdd
    build\lib.win32-3.8\_webp.pyd : fatal error LNK1120: 28 unresolved externals
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x86\\link.exe' failed with exit status 1120
    Error in atexit._run_exitfuncs:
    Traceback (most recent call last):
      File "C:\Users\silve\AppData\Roaming\Python\Python38\site-packages\colorama\ansitowin32.py", line 59, in closed
        return stream.closed
    ValueError: underlying buffer has been detached
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'c:\users\silve\appdata\local\programs\python\python38-32\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\silve\\AppData\\Local\\Temp\\pip-install-42g1k2k6\\webp_fb85ab40d19c4d06afd812d2782552ff\\setup.py'"'"'; __file__='"'"'C:\\Users\\silve\\AppData\\Local\\Temp\\pip-install-42g1k2k6\\webp_fb85ab40d19c4d06afd812d2782552ff\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\silve\AppData\Local\Temp\pip-record-niqhuezq\install-record.txt' --single-version-externally-managed --compile --install-headers 'c:\users\silve\appdata\local\programs\python\python38-32\Include\webp' Check the logs for full command output.
  `

**Comment by C-H-Simpson** [comment]
I'm trying to follow these instructions to install webp on Windows 10, and am getting the error
```
conans.errors.ConanException: compiler not defined for compiler.version
Please define compiler value first too
```

I tried following the instructions 
```
conan profile update "settings.compiler=Visual Studio" default
conan profile update "settings.compiler.version=14" default
pip install webp
```

But afterwards the resulting error messages are exactly the same.

Here is the full trace.
```
>pip install webp
Collecting webp
  Using cached webp-0.1.3.tar.gz (10 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\python.exe' 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' prepare_metadata_for_build_wheel 'C:\Users\ucbqc38\AppData\Local\Temp\tmps62z4eau'
       cwd: C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_abbbaa5c80284d85952669dbe241fc92
  Complete output (47 lines):
  platform.architecture: ('64bit', 'WindowsPE')
  platform.machine: AMD64
  Traceback (most recent call last):
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 363, in <module>
      main()
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py", line 164, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\build_meta.py", line 174, in prepare_metadata_for_build_wheel
      self.run_setup()
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\build_meta.py", line 158, in run_setup
      exec(compile(code, __file__, 'exec'), locals())
    File "setup.py", line 10, in <module>
      setup(
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 109, in setup
      _setup_distribution = dist = klass(attrs)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\dist.py", line 459, in __init__
      _Distribution.__init__(
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 293, in __init__
      self.finalize_options()
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\dist.py", line 837, in finalize_options
      ep(self)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\setuptools\dist.py", line 858, in _finalize_setup_keywords
      ep.load()(self, ep.name, value)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\cffi\setuptools_ext.py", line 219, in cffi_modules
      add_cffi_module(dist, cffi_module)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\cffi\setuptools_ext.py", line 49, in add_cffi_module
      execfile(build_file_name, mod_vars)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\cffi\setuptools_ext.py", line 25, in execfile
      exec(code, glob, glob)
    File "webp_build/builder.py", line 21, in <module>
      conan.install(path=getcwd(), cwd=tmp_dir, settings=settings, build=['missing'])
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
      return f(api, *args, **kwargs)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\conans\client\conan_api.py", line 597, in install
      graph_info = get_graph_info(profile_host, profile_build, cwd, None,
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\conans\client\conan_api.py", line 1560, in get_graph_info
      phost.process_settings(cache)
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\conans\model\profile.py", line 51, in process_settings
      self.processed_settings.values = Values.from_list(list(self.settings.items()))
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-build-env-gcgvgf1z\overlay\Lib\site-packages\conans\model\values.py", line 90, in from_list
      raise ConanException("%s not defined for %s\n"
  conans.errors.ConanException: compiler not defined for compiler.version
  Please define compiler value first too
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/73/52/070ca09e97489071d175cc880c8d738313abcf310e2edb6236af9e9a2828/webp-0.1.3.tar.gz#sha256=d06d6020c2b99f7895400d4d45fb33ebfea0f56badd2e63c6d38119ec6a9a38f (from https://pypi.org/simple/webp/) (requires-python:>=3.6). Command errored out with exit status 1: 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\python.exe' 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' prepare_metadata_for_build_wheel 'C:\Users\ucbqc38\AppData\Local\Temp\tmps62z4eau' Check the logs for full command output.
  Using cached webp-0.1.2.tar.gz (11 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_6ff3160ad36340fd8302b2904068b096\\setup.py'"'"'; __file__='"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_6ff3160ad36340fd8302b2904068b096\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\ucbqc38\AppData\Local\Temp\pip-pip-egg-info-c2nt5vff'
       cwd: C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_6ff3160ad36340fd8302b2904068b096\
  Complete output (38 lines):
  C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_6ff3160ad36340fd8302b2904068b096\setup.py", line 6, in <module>
      setup(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\core.py", line 109, in setup
      _setup_distribution = dist = klass(attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 459, in __init__
      _Distribution.__init__(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\dist.py", line 293, in __init__
      self.finalize_options()
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 837, in finalize_options
      ep(self)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 858, in _finalize_setup_keywords
      ep.load()(self, ep.name, value)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 219, in cffi_modules
      add_cffi_module(dist, cffi_module)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 49, in add_cffi_module
      execfile(build_file_name, mod_vars)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 25, in execfile
      exec(code, glob, glob)
    File "webp_build/builder.py", line 14, in <module>
      conan.install(path=getcwd(), cwd=tmp_dir)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
      return f(api, *args, **kwargs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 597, in install
      graph_info = get_graph_info(profile_host, profile_build, cwd, None,
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 1560, in get_graph_info
      phost.process_settings(cache)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\profile.py", line 51, in process_settings
      self.processed_settings.values = Values.from_list(list(self.settings.items()))
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\values.py", line 90, in from_list
      raise ConanException("%s not defined for %s\n"
  conans.errors.ConanException: compiler not defined for compiler.version
  Please define compiler value first too
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/16/60/be7e03cd6423308a5e7bbeace6e7eb612172534734043d8ffbae77862df3/webp-0.1.2.tar.gz#sha256=3d71b890ffdf39dcda02086f28fe66727f06b3a4854ec8e86d1318628c3ff044 (from https://pypi.org/simple/webp/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached webp-0.1.1.tar.gz (10 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_6951a7b2a57848f6ad5cef91c6a334c8\\setup.py'"'"'; __file__='"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_6951a7b2a57848f6ad5cef91c6a334c8\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\ucbqc38\AppData\Local\Temp\pip-pip-egg-info-gmb3ffai'
       cwd: C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_6951a7b2a57848f6ad5cef91c6a334c8\
  Complete output (38 lines):
  C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_6951a7b2a57848f6ad5cef91c6a334c8\setup.py", line 6, in <module>
      setup(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\core.py", line 109, in setup
      _setup_distribution = dist = klass(attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 459, in __init__
      _Distribution.__init__(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\dist.py", line 293, in __init__
      self.finalize_options()
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 837, in finalize_options
      ep(self)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 858, in _finalize_setup_keywords
      ep.load()(self, ep.name, value)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 219, in cffi_modules
      add_cffi_module(dist, cffi_module)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 49, in add_cffi_module
      execfile(build_file_name, mod_vars)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 25, in execfile
      exec(code, glob, glob)
    File "webp_build/builder.py", line 14, in <module>
      conan.install(path=getcwd(), cwd=tmp_dir)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
      return f(api, *args, **kwargs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 597, in install
      graph_info = get_graph_info(profile_host, profile_build, cwd, None,
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 1560, in get_graph_info
      phost.process_settings(cache)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\profile.py", line 51, in process_settings
      self.processed_settings.values = Values.from_list(list(self.settings.items()))
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\values.py", line 90, in from_list
      raise ConanException("%s not defined for %s\n"
  conans.errors.ConanException: compiler not defined for compiler.version
  Please define compiler value first too
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/58/90/44ac4929b2f293a58741876a177bf62f334c45bef4d0adc43b713203a1c1/webp-0.1.1.tar.gz#sha256=f4391dd6f56c1eef9df54071240ce0f3cf9bfb64320a4c8cf97d8a40bfd0eac9 (from https://pypi.org/simple/webp/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached webp-0.1.0.tar.gz (10 kB)
  Preparing metadata (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: 'C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_47a9347bdfd2410a8bd410da41084bfa\\setup.py'"'"'; __file__='"'"'C:\\Users\\ucbqc38\\AppData\\Local\\Temp\\pip-install-hvg0u1fg\\webp_47a9347bdfd2410a8bd410da41084bfa\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base 'C:\Users\ucbqc38\AppData\Local\Temp\pip-pip-egg-info-hsg_goy1'
       cwd: C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_47a9347bdfd2410a8bd410da41084bfa\
  Complete output (38 lines):
  C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
    warnings.warn(
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "C:\Users\ucbqc38\AppData\Local\Temp\pip-install-hvg0u1fg\webp_47a9347bdfd2410a8bd410da41084bfa\setup.py", line 6, in <module>
      setup(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\core.py", line 109, in setup
      _setup_distribution = dist = klass(attrs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 459, in __init__
      _Distribution.__init__(
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\_distutils\dist.py", line 293, in __init__
      self.finalize_options()
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 837, in finalize_options
      ep(self)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\setuptools\dist.py", line 858, in _finalize_setup_keywords
      ep.load()(self, ep.name, value)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 219, in cffi_modules
      add_cffi_module(dist, cffi_module)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 49, in add_cffi_module
      execfile(build_file_name, mod_vars)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\cffi\setuptools_ext.py", line 25, in execfile
      exec(code, glob, glob)
    File "webp_build/builder.py", line 14, in <module>
      conan.install(path=getcwd(), cwd=tmp_dir)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 93, in wrapper
      return f(api, *args, **kwargs)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 597, in install
      graph_info = get_graph_info(profile_host, profile_build, cwd, None,
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\client\conan_api.py", line 1560, in get_graph_info
      phost.process_settings(cache)
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\profile.py", line 51, in process_settings
      self.processed_settings.values = Values.from_list(list(self.settings.items()))
    File "C:\Users\ucbqc38\Documents\RoofPedia\Roofpedia\env\lib\site-packages\conans\model\values.py", line 90, in from_list
      raise ConanException("%s not defined for %s\n"
  conans.errors.ConanException: compiler not defined for compiler.version
  Please define compiler value first too
  ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/62/df/1efb36189ee596f6d3773c8c1b051ada90d244a601b0e4de00181e8fa6f9/webp-0.1.0.tar.gz#sha256=9f15bf22999789b05f12dd02e0970d44adf506e086ea9558e6c5abc4c5585126 (from https://pypi.org/simple/webp/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement webp (from versions: 0.1.0a4, 0.1.0a5, 0.1.0a6, 0.1.0a7, 0.1.0a9, 0.1.0a10, 0.1.0a11, 0.1.0a12, 0.1.0a13, 0.1.0a14, 0.1.0a15, 0.1.0a16, 0.1.0, 0.1.1, 0.1.2, 0.1.3)
ERROR: No matching distribution found for webp
```

**Comment by YISHISHEN** [comment]
sorry to bother, I'm trying to install the webp but always stuck in "Preparing wheel metadata... \"
and it's also fail in pycharm, could you please help me about this issue

**Comment by ArtyomEnRage** [comment]
Hello I'm still having this issue in 2022. What visual studio has to do with all this ? Why can't the install be simple and download the module like all others ? Does anyone has a zip of it ?

**Comment by anibali** [comment]
@ArtyomEnRage  There are Windows binary builds available for Python 3.10, 3.9, 3.8, 3.7, and 3.6 as can be seen here: https://pypi.org/project/webp/#files. If you do not have a setup which matches an available binary build, the C extension has to be built and this requires a compiler (on Windows this means Visual Studio). **Which version of Python are you using?**

> Why can't the install be simple and download the module like all others ?

Because this is a binding to a native library. Unlike pure Python packages, distributing pywebp so that it is "simple" to install means building for each different platform and each different Python version. I am just one person maintaining this in my free time.

As an aside, please try to be a bit more considerate and kind with your wording in the future. There are many developers just like me who are providing software to others for free in our spare time just to be helpful, not for pay. We are much more likely to spend time resolving issues with users who are understanding and work with us.
